appinventor/components/src/com/google/appinventor/components/runtime/Chart.java

```diff
@@ -0,0 +1,276 @@
+package com.google.appinventor.components.runtime;
+
+import android.app.Activity;
+import android.view.View;
+import android.widget.RelativeLayout;
+import com.google.appinventor.components.annotations.*;
+import com.google.appinventor.components.common.ComponentCategory;
+import com.google.appinventor.components.common.ComponentConstants;
+import com.google.appinventor.components.common.PropertyTypeConstants;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+
+@SimpleObject
+@DesignerComponent(version = 1,
+        category = ComponentCategory.CHARTS,
+        description = "A component that allows visualizing data")
+@UsesLibraries(libraries = "mpandroidchart.jar")
+public class Chart extends AndroidViewComponent implements ComponentContainer {
+    // Root layout of the Chart view. This is used to make Chart
+    // dynamic removal & adding easier.
+    private RelativeLayout view;
+
+    // Underlying Chart view
+    private ChartView chartView;
+
+    private int type;
+    private int backgroundColor;
+    private String description;
+
+    // Attached Data components
+    private ArrayList<ChartDataBase> dataComponents;
+
+    /**
+     * Creates a new Chart component.
+     *
+     * @param container container, component will be placed in
+     */
+    public Chart(ComponentContainer container) {
+        super(container);
+
+        view = new RelativeLayout(container.$context());
```

Comment:
The underlying view for the Chart is a RelativeLayout, so that Chart views can easily be added and removed dynamically. Is this a good approach, or should something else be done here?